pkg/controller/servicebindingrequest/retriever.go

```diff
@@ -2,27 +2,32 @@ package servicebindingrequest
 
 import (
 	"context"
+	"encoding/base64"
 	"fmt"
 	"strings"
 
 	"github.com/go-logr/logr"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
 // Retriever reads all data referred in plan instance, and store in a secret.
 type Retriever struct {
-	ctx           context.Context   // request context
-	client        client.Client     // Kubernetes API client
-	plan          *Plan             // plan instance
-	logger        logr.Logger       // logger instance
-	data          map[string][]byte // data retrieved
-	volumeKeys    []string
-	bindingPrefix string
+	logger        logr.Logger                  // logger instance
+	data          map[string][]byte            // data retrieved
+	objects       []*unstructured.Unstructured // list of objects employed
+	ctx           context.Context              // request context
+	client        dynamic.Interface            // Kubernetes API client
+	plan          *Plan                        // plan instance
+	volumeKeys    []string                     // list of keys found
+	bindingPrefix string                       // prefix for variable names
```

Comment:
Shall we keep ordered fields on structs? This would allow one to scan better this sort of diff scenarios.

Reply:
That's a good question. The approach is not standardized throughout the project, so would be good to document this convention and re-asses those again.